packages/rs-drive/src/drive/cache/protocol_version.rs

```diff
@@ -56,33 +55,30 @@ impl ProtocolVersionsCache {
 
     /// Merge block cache to global cache
     pub fn merge_block_cache(&mut self) {
-        if self.needs_wipe {
-            self.global_cache.clear();
-            self.block_cache.clear();
-            self.needs_wipe = false;
-        } else {
-            self.global_cache.extend(self.block_cache.drain());
-        }
+        self.global_cache.extend(self.block_cache.drain());
+    }
+
+    /// Clears the cache
+    pub fn clear(&mut self) {
+        self.loaded = false;
+        self.global_cache.clear();
+        self.block_cache.clear();
     }
 
     /// Clear block cache
     pub fn clear_block_cache(&mut self) {
         self.block_cache.clear()
     }
 
-    /// Versions passing threshold
-    pub fn versions_passing_threshold(
-        &mut self,
-        required_upgraded_hpns: u64,
-    ) -> Vec<ProtocolVersion> {
-        self.needs_wipe = true;
+    /// Collect versions passing threshold
+    pub fn versions_passing_threshold(&self, required_upgraded_hpmns: u64) -> Vec<ProtocolVersion> {
         let mut cache = self.global_cache.clone();
 
-        cache.extend(self.block_cache.drain());
+        cache.extend(self.block_cache.iter());
```

Comment:
I think drain is better so we don't need to clone elements.

Reply:
In this case this method will alter cache. Getter shouldn't alter cache. How many elements you expect in block cache? I think with such small number there is no difference and it happen only on epoch change